### ISSUE TITLE
Error loading from classpath within .zip files

### ISSUE THREAD

**Issue by ernestojpg** [eclipse/vert.x#1316]
### Issue description

I have found an issue with the `vertx.fileSystem().readFile()` function.

The function will fail if you try to load a file from the classpath and that file is stored inside a .zip file.

The problem is in the class `io.vertx.core.impl.FileResolver`. The method `unpackFromJarURL()` is assuming that the file will be always stored inside a .jar file:

```
  private synchronized File unpackFromJarURL(URL url, String fileName, ClassLoader cl) {
    try {
      ZipFile zip;
      String path = url.getPath();
      int idx1 = path.lastIndexOf(".jar!");
      ...
```

The method will throw an `ArrayIndexOutOfBoundsException` if the file is stored inside a .zip file.

### How to reproduce it

I have attached two files:
- **[test.zip](https://github.com/eclipse/vert.x/files/147962/test.zip)**: Compressed file that implements a Vert.x service. It contains two files:
  - mainVerticle.js: Main Service Verticle, written in JavaScript, that try to load a file.
  - myMod.json: Service descriptor, that indicates the location of the main service verticle.
- **[vertx-service-factory-3.2.1.zip](https://github.com/eclipse/vert.x/files/147964/vertx-service-factory-3.2.1.zip)**: Official Vert.x Service Factories implementation, used for loading the above service. Note that the .zip extension is not relevant here. Renamed from .jar to .zip just for being able to upload the file here.

For reproducing the issue just download both files and introduced the following commands (Linux/Mac):
- `export CLASSPATH=${CLASSPTH}:.:vertx-service-factory-3.2.1.zip:test.zip`
- `vertx run service:myMod`

You should see this log:
```
Hello from the mainVerticle.js
Succeeded in deploying verticle
Unhandled exception
<eval>#88:17<eval>:7:7 Error loading myMod.json file: java.lang.ArrayIndexOutOfBoundsException: 1
	at jdk.nashorn.internal.runtime.ECMAException.create(ECMAException.java:113)
	at jdk.nashorn.internal.scripts.Script$Recompilation$138$274AA$\^eval\_#88\!17\^eval\_.L:1$L:5(mainVerticle.js:7)
	at jdk.nashorn.internal.scripts.Script$Recompilation$137$31550A$\^eval\_#88\!17\^eval\_.L:1$FileSystem$readFile$L:831(vertx-js/file_system.js:835)
	at io.vertx.core.Handler$$NashornJavaAdapter.handle(Unknown Source)
	at io.vertx.core.impl.FutureImpl.checkCallHandler(FutureImpl.java:135)
	at io.vertx.core.impl.FutureImpl.setHandler(FutureImpl.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$null$14(ContextImpl.java:305)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$16(ContextImpl.java:335)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:358)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:357)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by ernestojpg** [comment]
I have submitted a quick fix for the issue in pull request #1317

**Comment by vietj** [comment]
done
